### DISCUSSION TITLE
Is there a `DateTimePicker`?

### DISCUSSION THREAD

**Discussion by karmeye** [CommunityToolkit/Labs-Windows#538] [Ideas]
I've only found separate controls for date and time, not a combined one like this one:

![Screenshot 1](https://github.com/CommunityToolkit/Windows/assets/366038/3684ff83-ac36-4c56-885f-4ae666f253b3)

**Comment by michael-hawker** [comment]
Hi @karmeye, moved this to Labs as an idea. We don't have one in the toolkit. They're just two separate control in the platform, so I think they're just bundled together here or this is a custom experience for this dialog rather.

Syncfusion has one for UWP, but also appears to be both `DatePicker` and `TimePicker` bundled together, [their docs here](https://help.syncfusion.com/uwp/datetimepicker/overview):

![image](https://github.com/CommunityToolkit/Labs-Windows/assets/24302614/d5d1061c-db6a-4a90-908f-a4265399d822)

Seems like WindowsForms has something called `DateTimePicker`, but may just be a Date Picker that returns a `DateTime` value. I couldn't find a picture of it in the WinForms docs...

Could be interesting to have something combined here which can have some options for display (like do you want the dropdown that opens a calendar or the individual date selection as the normal DatePicker, etc...). Then it just returns the single value for the developer to worry about vs. having to merge two together.

References:
- WinUI: https://learn.microsoft.com/windows/apps/design/controls/date-picker
- WinUI: https://learn.microsoft.com/windows/apps/design/controls/time-picker
- WinForms: https://learn.microsoft.com/dotnet/api/system.windows.forms.datetimepicker
- Syncfusion: https://help.syncfusion.com/uwp/datetimepicker/overview

**Reply by karmeye** [reply]
Thanks for the reply @michael-hawke!

I realize now that the image attached is what I currently have as a workaround, it's two controls.

> Seems like WindowsForms has something called DateTimePicker,

Yes, it was actually the `DateTimePicker` of WinForms I am used to, coming from WinForms, expecting to have one in WinUI. Yes it returns a [DateTime](https://learn.microsoft.com/en-us/dotnet/api/system.windows.forms.datetimepicker.value?view=windowsdesktop-8.0).

> Could be interesting to have something combined here which can have some options for display (like do you want the dropdown that opens a calendar or the individual date selection as the normal DatePicker, etc...). Then it just returns the single value for the developer to worry about vs. having to merge two together.

Yes, I agree. There's not many options. And a calendar would be expected in some cases. Also it introduces some ambiguity as there is now two date pickers - DatePicker and CalendarView. 

I think my issue was that I prefer the user to be able to set both date and time in one flyout, but actually I think the current solution is fine too.

**Reply by karmeye** [reply]
Btw, @michael-hawker could you move [this one](https://github.com/CommunityToolkit/Windows/discussions/351) to labs too?
